src/rdkafka.c

```diff
@@ -270,7 +270,51 @@ void rd_kafka_log0 (const rd_kafka_conf_t *conf,
         rd_kafka_log_buf(conf, rk, level, fac, buf);
 }
 
+void rd_kafka_oauthbearer_token_refresh_success(rd_kafka_t *rk,
+                const char *token_value, int64_t md_lifetime_ms,
+                const char *md_principal_name, int64_t md_start_time_ms) {
```

Comment:
The Java client has an optional start time metadata.  That isn't a reason to have it here, of course.  The Java client uses the value to calculate when to do the next refresh.  For example, let's say the refresh is supposed to happen at the 80% mark.  That 80% mark is defined to be the 80% point of the token's lifetime.  So if the token start time was 1:00 and the token end time is 2:00 the refresh point would be 1:48.  The start time is optional, and if it isn't present then the algorithm uses the current time instead.  So if the current time is 1:01 and there is no start time available then the refresh time would be in (80% * 59 minutes) from now.  Frankly, now that you ask about it, I don't think in reality there is much value in tracking the start time because tokens won't be created in the distant past -- the start time will always be very close to the current time.  If someone had asked the same question when I was doing KIP-255 I probably would have realized it wasn't necessary.  I will remove it and always use the current time when calculating the refresh point.